hls-plugin-api/src/Ide/Types.hs

```diff
@@ -220,21 +221,22 @@ data CustomConfig = forall r. CustomConfig (Properties r)
 -- which can be inferred from handlers registered by the plugin.
 -- @config@ is called custom config, which is defined using 'Properties'.
 data ConfigDescriptor = ConfigDescriptor {
-  -- | Whether or not to generate generic configs.
-  configEnableGenericConfig :: Bool,
+  -- | Initial values for the generic config, if the plugin admits one
+  configInitialGenericConfig :: Maybe PluginConfig,
   -- | Whether or not to generate @diagnosticsOn@ config.
   -- Diagnostics emit in arbitrary shake rules,
   -- so we can't know statically if the plugin produces diagnostics
-  configHasDiagnostics      :: Bool,
+  configHasDiagnostics       :: Bool,
   -- | Custom config.
-  configCustomConfig        :: CustomConfig
+  configCustomConfig         :: CustomConfig
 }
 
 mkCustomConfig :: Properties r -> CustomConfig
 mkCustomConfig = CustomConfig
 
 defaultConfigDescriptor :: ConfigDescriptor
-defaultConfigDescriptor = ConfigDescriptor True False (mkCustomConfig emptyProperties)
+defaultConfigDescriptor =
+    ConfigDescriptor (Just Data.Default.def) False (mkCustomConfig emptyProperties)
```

Comment:
Hmm, this surprises me, I'd have expected the default value to be `Nothing` i.e. the plugin doesn't care?